src/test/java/org/opensearch/knn/index/OpenSearchIT.java

```diff
@@ -399,4 +399,46 @@ public void testExistsQuery() throws IOException {
 
         assertEquals(3, parseTotalSearchHits(EntityUtils.toString(response.getEntity())));
     }
+
+    public void testIndexingVectorValidation_updateVectorWithNull() throws Exception {
+        Settings settings = Settings.builder()
+                .put(getKNNDefaultIndexSettings())
+                .build();
+
+        createKnnIndex(INDEX_NAME, settings, createKnnIndexMapping(FIELD_NAME, 4));
+
+        // valid case with 4 dimension
```

Comment:
shall we add test case with two documents and k =1, where after update, you will get other document as search response?

Reply:
I think such test case makes sense, I can extend existing test case and start with two documents instead of one, then after update search will return the other document that wasn't updated. Part with retrieving updated document by id should remains the same

Reply:
addressed your comment in next commit - https://github.com/opensearch-project/k-NN/pull/237/commits/a038333b8ac8d982a383234336b8ac7dea5b02c3, please check

Reply:
Thank you!